src/lib/summary.ts

```diff
@@ -0,0 +1,269 @@
+   * @param rangesTuple1 The rangesTuple where the ranges will be extended.
+   * @param rangesTuple2 The rangesTuple where the ranges will be used to extend.
+   */
+  public extendRangesTupleWithRangesTuple(rangesTuple1: RangesTuple, rangesTuple2: RangesTuple) {
+    if (!this.canExtentRangesTupleWithRangesTuple(rangesTuple1, rangesTuple2)) {
+      throw new RangeError("a value in the rangeTuple could not be extended");
+    }
+
+    rangesTuple1[0].extendWithRange(rangesTuple2[0]);
+    rangesTuple1[1].extendWithRange(rangesTuple2[1]);
+  }
+
+  /**
+   * Converts a list of matching lines to a list of ranges.
+   * @param matches The list of matching lines.
+   * @returns A list of tuples that contains two ranges, where the frist and second range correspond to the line
+   * numbers of each file.
+   */
+  public matchesToRange(matches: Array<[number, number]>): RangesTuple[] {
+    let ranges: RangesTuple[] = new Array();
+
+    matches.forEach(next => {
+      const rangeTupleIndex: number = ranges.findIndex(rangeTuple => {
+        return (
+          rangeTuple[0].canExtendWithNumber(next[0], this.gapSize) &&
+          rangeTuple[1].canExtendWithNumber(next[1], this.gapSize)
+        );
+      });
+      if (rangeTupleIndex === -1) {
+        ranges.push([new Range(next[0], next[0]), new Range(next[1], next[1])]);
+      } else {
+        ranges[rangeTupleIndex][0].extendWithNumber(next[0]);
+        ranges[rangeTupleIndex][1].extendWithNumber(next[1]);
+      }
+    });
+
+    ranges = this.concatenateRanges(ranges);
+
+    return this.filterByMinimumLines(ranges);
+  }
+
+  /**
+   * Remove all ranges that only contain less the minimum required lines.
+   * @param rangesTupleArray The rangesTupleArray you want filter.
+   */
+
+  public filterByMinimumLines(rangesTupleArray: RangesTuple[]): RangesTuple[] {
+    return rangesTupleArray.filter(
+      rangesTuple =>
+        Math.max(rangesTuple[0].getLineCount(), rangesTuple[1].getLineCount()) >=
+          this.minimumMaximumLines &&
+        Math.min(rangesTuple[0].getLineCount(), rangesTuple[1].getLineCount()) >=
+          this.minimumMinimumLines,
+    );
+  }
+
+  /**
+   * Concatenates all the rangesTuples whenever possible.
+   * @param rangesTupleArray The rangesTuples you want to concatenate where possible. Returns a new array with the
+   * result.
+   */
+  public concatenateRanges(rangesTupleArray: RangesTuple[]): RangesTuple[] {
+    const ranges: RangesTuple[] = rangesTupleArray.slice();
+    // If two rangesTuples overlap with each other then extend the second with the first and remove the
+    // first from the array.
+    for (let i = ranges.length - 1; i >= 0; i--) {
+      for (let j = i; j >= 0; j--) {
+        if (i !== j) {
+          if (this.canExtentRangesTupleWithRangesTuple(ranges[i], ranges[j])) {
+            this.extendRangesTupleWithRangesTuple(ranges[j], ranges[i]);
+            ranges.splice(i, 1);
+          }
+          break;
```

Comment:
I can be mistaken, but shouldn't this break be in the if? If you extended the range, the i'th element is removed, so the outer loop should start a new iteration with a new i.

Reply:
You are correct, I fixed this in the feature/CLI-2 branch and added a test for it but forgot to make the same changes in this branch.